core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java

```diff
@@ -251,7 +250,19 @@ public ServerBuilder port(InetSocketAddress localAddress, SessionProtocol protoc
      * @see <a href="#no_port_specified">What happens if no HTTP(S) port is specified?</a>
      */
     public ServerBuilder port(ServerPort port) {
-        ports.add(requireNonNull(port, "port"));
+        requireNonNull(port, "port");
+        for (int i = 0; i < ports.size(); i++) {
+            final ServerPort p = ports.get(i);
+            if (p.equals(port)) {
```

Comment:
Could we compare only the port number here and make `ServerPort.equals()` returns when both port number and protocols match?